pkg/storage/util/util.go

```diff
@@ -50,18 +53,47 @@ func UpdateCondition(cr *imageregistryv1.Config, conditionType string, status op
 	cr.Status.Conditions = conditions
 }
 
-func GetClusterVersionConfig(kubeconfig *rest.Config) (*configv1.ClusterVersion, error) {
-	client, err := configv1client.NewForConfig(kubeconfig)
+// GetInfrastructure gets information about the cloud platform that the cluster is
+// installed on including the Type, Region, and other platform specific information.
+// Currently the install config is used as a backup to be compatible with upgrades
+// from 4.1 -> 4.2 when platformStatus did not exist, but should be able to be removed
+// in the future.
+func GetInfrastructure(listers *regopclient.Listers) (*configv1.Infrastructure, error) {
+	infra, err := listers.Infrastructures.Get("cluster")
 	if err != nil {
 		return nil, err
 	}
-	return client.ConfigV1().ClusterVersions().Get("version", metav1.GetOptions{})
+
+	if infra.Status.PlatformStatus == nil {
+		infra.Status.PlatformStatus = &configv1.PlatformStatus{
+			Type: infra.Status.Platform,
+		}
+
+		// TODO: Eventually we should be able to remove our dependency on the install config
+		// but it is needed for now since platformStatus doesn't get set on upgrade
+		// from 4.1 -> 4.2
+		ic, err := listers.InstallerSecrets.Get("cluster-config-v1")
```

Comment:
I can see this being not guaranteed for during a 4.1 -> 4.2 upgrade, but we should be prepared for this secret to disappear for 4.2 -> 4.3 upgrades

Reply:
Once PlatformStatus is guaranteed to be populated, that should be fine.